Empresas/Controller/EnterpriseDetailViewController.swift

```diff
@@ -7,31 +7,39 @@
 
 import UIKit
 
-class EnterpriseDetailViewController: UIViewController, CustomTellerBack {
+class EnterpriseDetailViewController: BaseViewController {
     
+    private let presenter: EnterpriseDetailPresenting
     private var enterprise: Enterprise?
+    private lazy var enterpriseDetailView = EnterpriseDetailView(enterprise: nil)
     
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view.
+        setupCustomNavigation()
     }
     
     override func loadView() {
-        let enterpriseView = EnterpriseDetailView(enterprise: self.enterprise)
-        enterpriseView.delegate = self
-        view = enterpriseView
-    }
-    
-    func backButtonClicked(_: UIButton) {
-        self.navigationController?.popViewController(animated: true)
+        enterpriseDetailView.delegate = self
+        view = enterpriseDetailView
     }
 
-    init(enterprise: Enterprise) {
-        super.init(nibName: nil, bundle: nil)
+    init(presenter: EnterpriseDetailPresenting, enterprise: Enterprise) {
+        self.presenter = presenter
         self.enterprise = enterprise
+        super.init(nibName: nil, bundle: nil)
+        self.enterpriseDetailView = EnterpriseDetailView(enterprise: self.enterprise)
```

Comment:
E aqui vc pode tirar essa linha, pq fica repetido. O lazy var vai instanciar a variável enterpriseDetailView só no momento que for chamada, e isso faz com que a variável tenha acesso as propriedades da view controller atualizadas